### PR TITLE
feat(diagnostic): add `showRelatedInformation`

### PR DESCRIPTION
**Pull request by hexh250786313**
See: https://github.com/neoclide/coc.nvim/issues/5086

alt text
alt text

### CODE REVIEW

**Comment by fannheyward** [comment]
screenshot please, for better understanding.

**Comment by hexh250786313** [comment]
@fannheyward Screenshots added.

**Comment by fannheyward** [comment]
Shall we remove the `separateRelatedInformationAsDiagnostics`? This new configuration controls to display related in floating window, I'd prefer to remove the old configuration, which only separates them but doesn't shows in relation.

**Comment by hexh250786313** [comment]
@fannheyward Removing the `separateRelatedInformationAsDiagnostics` feature is ok, the separated diagnostic doesn't really provide much useful information:

image

If we need to find the related diagnostic, we can just use `:CocCommand workspace.diagnosticRelated`.

Moreover, for some reason, the `separateRelatedInformationAsDiagnostics` only works for customized language services configured by `coc-settings.json`, not for some of plug-in language servers (e.g. coc-tsserver). See: https://github.com/neoclide/coc.nvim/discussions/5078

I will make a commit for it shortly.

**Comment by fannheyward** [comment]
> the separated diagnostic doesn't really provide much useful informatio

Yes, this is what I said not showing in relation.

File: src/language-client/client.ts [hunk 1]
```diff
@@ -1575,24 +1568,8 @@ export abstract class BaseLanguageClient implements FeatureClient<Middleware, La
   private setDiagnostics(uri: string, diagnostics: Diagnostic[] | undefined) {
     if (!this._diagnostics) return
 
-    const separate = this.clientOptions.separateDiagnostics
     // TODO make is async
```
Comment (fannheyward) [inline src/language-client/client.ts:1571]: remove this comment too.

**Review by fannheyward — CHANGES_REQUESTED** [review]

File: src/language-client/client.ts (hunk 1 above)
Comment (hexh250786313) [reply src/language-client/client.ts]: Done.

**Review by fannheyward — APPROVED** [review]

**Pull request merged**